### PR TITLE
feat(sync-translated-content): gather releated upstream commits

### PR DESCRIPTION
**Pull request by yin1999**
### Description

gather releated upstream commits

---

Tested in my fork, check branch: https://github.com/yin1999/translated-content/tree/test-gather-urls

workflow run: https://github.com/yin1999/translated-content/actions/runs/8182413701

screenshot:

image

image

### Related issues and pull requests

https://github.com/mdn/translated-content/pull/9199#discussion_r992885708

### CODE REVIEW

**Comment by yin1999** [comment]
ping @OnkarRuikar, @bsmth for a review :)

File: .github/workflows/sync-translated-content.yml [hunk 1]
```diff
@@ -59,6 +59,36 @@ jobs:
...
 
+      - name: Gather related upstream commits
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # get upstream head sha
+          UPSTREAM_HEAD_SHA=$(cd "${{ github.workspace }}/mdn/content" && git rev-parse HEAD)
+          echo "upstream head sha: ${UPSTREAM_HEAD_SHA}"
+          # get original path of moved files
+          MOVED_FILES=$(git diff --cached --name-status --diff-filter=R | cut -f2 | sed "0,/${{ matrix.lang }}/s//en-us/")
```
Comment (OnkarRuikar) [inline .github/workflows/sync-translated-content.yml:70]: Could you explain why the pattern `0,/ja/s//en-us/`? In my local it changed only the first line of the output.

Can we have a simpler version:
```suggestion
          MOVED_FILES=$(git diff --cached --name-status --diff-filter=R | cut -f2 | sed "s|files/${{ matrix.lang }}|files/en-us|")
```

Comment (yin1999) [reply .github/workflows/sync-translated-content.yml]: my neglect! fixed in d78ef29982b98b07cfcdb5ae56bab29f72c90fed

File: .github/workflows/sync-translated-content.yml [hunk 2]
```diff
@@ -59,6 +59,36 @@ jobs:
...
+          UPSTREAM_HEAD_SHA=$(cd "${{ github.workspace }}/mdn/content" && git rev-parse HEAD)
+          echo "upstream head sha: ${UPSTREAM_HEAD_SHA}"
+          # get original path of moved files
+          MOVED_FILES=$(git diff --cached --name-status --diff-filter=R | cut -f2 | sed "s|files/${{ matrix.lang }}|files/en-us|")
+          FILE_COMMIT_URLS=""
+          while read -r MOVED_FILE; do
+            # gather related upstream commits
+            COMMIT_URL=$(gh api -XGET repos/mdn/content/commits -F path="${MOVED_FILE}" -F per_page=1 -F sha="${UPSTREAM_HEAD_SHA}" --jq '.[0].html_url')
+            if [ -n "${COMMIT_URL}" ]; then
+              FILE_COMMIT_URLS="${FILE_COMMIT_URLS}- ${COMMIT_URL}\n"
```
Comment (OnkarRuikar) [inline .github/workflows/sync-translated-content.yml:76]:
```suggestion
              FILE_COMMIT_URLS+="- ${COMMIT_URL}\n"
```

File: .github/workflows/sync-translated-content.yml [hunk 3]
```diff
@@ -59,6 +59,36 @@ jobs:
...
+            fi
+          done <<< "${MOVED_FILES}"
+
+          FILE_COMMIT_URLS=$(echo -e "${FILE_COMMIT_URLS}" | grep -v '^$' | sort | uniq)
```
Comment (OnkarRuikar) [inline .github/workflows/sync-translated-content.yml:80]:
```suggestion
          FILE_COMMIT_URLS=$(echo -e "${FILE_COMMIT_URLS}" | grep -v '^$' | uniq | sort)
```

**Review by OnkarRuikar — COMMENTED** [review]
nit

**Comment by OnkarRuikar** [comment]
For testing all the changes so far is it possible to do following in your forks main branch?
```
git reset --soft HEAD~1
git stash save "my workflow change"
git reset --hard 8fc08d671390ffb07761a3aeb31e6284ceeba20f
git stash pop
git commit -am "feat(sync-translated-content): gather releated upstream commits"
git push --force
```
Re-trigger the workflows on GitHub. This gives us more commit URLs to test.

To bring back the fork to normal in local as well as on github:
```
git reset --soft HEAD~1
git stash save "my workflow change"
git pull mdn-translated-content main
git push --force
```

**Comment by yin1999** [comment]
> For testing all the changes so far is it possible to do following in your forks main branch?
> 
> ```
> git reset --soft HEAD~1
> git stash save "my workflow change"
> git reset --hard 8fc08d671390ffb07761a3aeb31e6284ceeba20f
> git stash pop
> git commit -am "feat(sync-translated-content): gather releated upstream commits"
> git push --force
> ```
> 
> Re-trigger the workflows on GitHub. This gives us more commit URLs to test.
> 
> To bring back the fork to normal in local as well as on github:
> 
> ```
> git reset --soft HEAD~1
> git stash save "my workflow change"
> git pull mdn-translated-content main
> git push --force
> ```

Done in main branch, see workflow run: https://github.com/yin1999/translated-content/actions/runs/8182413701

**Review by OnkarRuikar — APPROVED** [review]
https://github.com/yin1999/translated-content/pull/41 looks great!

**Review by bsmth — APPROVED** [review]
Cool! Looks good! 👍🏻

**Comment by yin1999** [comment]
Thank you @OnkarRuikar, @bsmth. I'm going to merge this shortly and have it a test in this repo after :ship:

**Pull request merged**